### ISSUE TITLE
Further small improvements in light deflection calculations

### ISSUE THREAD

**Issue by mhvk** [astropy/astropy#10843]
In #10666, the light deflection calculation was updated to ensure that a source was not simply taken at infinite distance. In principle, as noted by @mkbrewer in https://github.com/astropy/astropy/pull/10666#issuecomment-678528759 (and following) and https://github.com/astropy/astropy/pull/10666#issuecomment-707369857, two further improvements are possible:
1. One should include that the deflecting source moved a bit in the time the light took to travel to the observer.
2. Other bodies than the Sun also lead to some light deflection (Jupiter at least should be easy to include with `ldn`).

Separately, we should document what exactly we do and do not include.

cc @mkbrewer, @StuartLittlefair

**Comment by mhvk** [comment]
I included a link to further discussion in https://github.com/astropy/astropy/pull/10666#issuecomment-707827131 above, in particular for a nice plot with residuals because of gravitational light bending of the Earth. @mkbrewer - would those in practice be most important for satellites? For most ground-based observatories, I'd think uncertainties in atmospheric/ionospheric effects are far larger. On the other hand, it doesn't hurt to include all effects one is aware of...

**Comment by mkbrewer** [comment]
Yes, indeed! For ground based observations, the uncertainty in refraction alone near the horizon is much larger than this effect. That said, if one wanted an absolutely rigorous airless model, this effect would be included.
